### ISSUE TITLE
Get stucked when train patchcore 

### ISSUE THREAD

**Issue by BirdFrozen** [openvinotoolkit/anomalib#282]
**Describe the bug**

change train_batch_size and test_batch_size to 4 in anomalib\models\padim\config.yaml

`python tools/train.py --model patchcore` 
shows
`Epoch 0:   0%|                                                                                                                         | 0/292 [00:00<?, ?it/s]C:\ProgramData\Anaconda3\envs\anomalib_env\lib\site-packages\pytorch_lightning\loops\optimization\optimizer_loop.py:137: UserWarning: `training_step` returned `None`. If this was on purpose, ignore this warning...
  self.warning_cache.warn("`training_step` returned `None`. If this was on purpose, ignore this warning...")
Epoch 0:  72%|████████████████2022-04-26 15:36:05,460 - anomalib.models.patchcore.lightning_model - INFO - Aggregating the embedding extracted from the training set.ation: 0it [00:00, ?it/s]
2022-04-26 15:36:05,633 - anomalib.models.patchcore.lightning_model - INFO - Applying core-set subsampling to get the embedding.`

and then stucked

**Screenshots**
![image](https://user-images.githubusercontent.com/98138576/165250340-eecf2caa-72a7-4402-a658-75607c660ef9.png)

**Hardware and Software Configuration**
- OS: Win10
- CUDA Version [e.g. 11.0]

**Additional context**
- I can run padim and get a good result.

**Comment by alexriedel1** [comment]
Depending on your dataset size and your computing ressources, the core-set subsampling can take quite a while. Also you shouldn't increase the test batch size as this will break the torchmetrics metric calculations.

**Comment by BirdFrozen** [comment]
Thanks a lot,I think it is very slow because anomalib install cpu version of pytorch,so I install a gpu version pytorch and it is very fast now.

**Comment by samet-akcay** [comment]
@BirdFrozen, now that you've fixed the issue, I'm closing this.

**Comment by 785256592** [comment]
> 

dear, @samet-akcay,  I used gpu version pytorch,but it also stucked in "2023-08-21 15:10:05,633 - anomalib.models.patchcore.lightning_model - INFO - Applying core-set subsampling to get the embedding."。the gpu was almost full.      20.6/24GB  RTX3090

**Comment by alexriedel1** [comment]
> > 
> 
> dear, @samet-akcay, I used gpu version pytorch,but it also stucked in "2023-08-21 15:10:05,633 - anomalib.models.patchcore.lightning_model - INFO - Applying core-set subsampling to get the embedding."。the gpu was almost full. 20.6/24GB RTX3090

It can take a very long time as I said before! How long have you been waiting? It might take an hour or so..

**Comment by samet-akcay** [comment]
@blaz-r, can you share your insights here in regards to the possible leaks and how to avoid it

**Comment by blaz-r** [comment]
@samet-akcay when training a single model, there shouldn't be any problems with leaks. As Alexander pointed out, it's just how Patchcore is implemented. Coreset subsampling can take really long. The huge memory consumption on the other hand comes from computing distances on many vectors, because it's done with matrix multiplication, meaning there is an entire matrix of all distances stored in memory.
I'm not sure how authors of the paper implemented this, but with the current anomalib it's like that and that requires a lot of memory.

**Comment by samet-akcay** [comment]
I think they are using [faiss](https://github.com/facebookresearch/faiss) to speed up the coreset sampling

**Comment by alexriedel1** [comment]
A quick fix in the mean time would be to use a tqdm in this loop, so people can see that something is happening https://github.com/openvinotoolkit/anomalib/blob/cdedf05451cc57fb373866595e8ef23a820cf123/src/anomalib/models/components/sampling/k_center_greedy.py#L103-L109
